laravel/app/Http/Controllers/ProductsController.php

```diff
@@ -152,7 +152,14 @@ public function addtocart(Request $request, Product $product)
           }
         }
 
+        // Count items in cart
+        $totalItems = 0;
+        foreach ($shoppingList as $p) {
```

Comment:
Instead of using a foreach, try using a map instead.

http://php.net/manual/en/function.array-map.php

If you don't like using php arrays, you could also use a collection and the collect helper:

https://laravel.com/docs/5.5/collections

Reply:
Actually a reduce would be better.